### PR TITLE
Improve contributing guidelines

### PR DESCRIPTION
**Pull request by cijothomas**
Hope they makes sense!

### CODE REVIEW

File: CONTRIBUTING.md
Comment (lalitb) [inline CONTRIBUTING.md:91]: we can decide on the number of approvals good enough to merge, depending on the size and other factors. Can be done separate of this PR.

**Review by lalitb — APPROVED** [review]

File: CONTRIBUTING.md
Comment (TommyCpp) [reply CONTRIBUTING.md:91]: Yeah I'd like enforce 2 approval before merging given we have more approvers

Comment (cijothomas) [reply CONTRIBUTING.md:91]: For now, maybe maintainers can decide to wait for 2 approvals before merging on a case-by-case basis, instead of requiring/enforcing it for every PRs (just to keep the momentum going!). Would this work?
(That is how many others SIGs are operating. But happy to revisit the idea for this SIG)

Comment (TommyCpp) [reply CONTRIBUTING.md:91]: Yeah we should call it out in documentation though. So that people will review the PR with 1 approval

Comment (cijothomas) [reply CONTRIBUTING.md:91]: Do we want to? I guess maintainers/approvers are already doing so.... Writing something like this in official doc, without being too restrictive is hard 😄

**Review by shaun-cox — APPROVED** [review]

File: CONTRIBUTING.md
Comment (cijothomas) [reply CONTRIBUTING.md:91]: https://github.com/open-telemetry/opentelemetry-rust/pull/1354/commits/6a53393c0fa233e3928bef877dbe7a89d6e0babd Added some wording! @TommyCpp see if that makes sense!

Comment (TommyCpp) [reply CONTRIBUTING.md:91]: Thanks for addressing that

**Review by TommyCpp — APPROVED** [review]

**Pull request merged**
